Add bookings

Chegulov opened the pull request.

Chegulov committed:
add-bookings
добавлены БД, jpa, querydsl. Рефакторинг кода под новое состояние.

Chegulov committed:
add-bookings
добавлена возможность бронирования и комментирования.

Chegulov committed:
add-bookings
добавлена возможность бронирования и комментирования.

Chegulov committed:
add-bookings
добавлена возможность бронирования и комментирования.

avlin87 commented on src/main/java/ru/practicum/shareit/booking/service/BookingServiceImpl.java:
```diff
@@ -0,0 +1,165 @@
+        userRepository.findById(userId)
+                .orElseThrow(() -> new DataNotFoundException("Пользователь с id=" + userId + " не найден."));
+
+        switch (state) {
```
Можно сократить код путем выделения общей части, смотри, пожалуйста, пример:

```
...{
...
    switch (state.toUpperCase()) {
        case "CURRENT":
            return mapToDtoOutputList(bookingRepository.readAllBookerCurrentBookings(userId, now));
        case "PAST":
            return mapToDtoOutputList(bookingRepository.readAllBookerPastBookings(userId, now));
...
        default:
            throw new WrongStateException("Unknown state: " + state);
    }
}

private List<BookingDtoOutput> mapToDtoOutputList(List<Booking> bookings) {
    return bookings.stream()
            .map(bookingMapper::createDtoOutput)
            .collect(Collectors.toList());
}
```

Chegulov replied:
добавил метод в маппер

avlin87 commented on src/main/java/ru/practicum/shareit/booking/service/BookingServiceImpl.java:
```diff
@@ -0,0 +1,165 @@
+                .map(Item::getId)
+                .collect(Collectors.toList());
+
+        switch (state) {
```
`Можно лучше`
Безопаснее будет привести `state` к toUpperCase.

Chegulov replied:
спасибо, поправил

avlin87 commented on src/main/java/ru/practicum/shareit/item/model/Comment.java:
```diff
@@ -0,0 +1,35 @@
+    private Long id;
+
+    @Column(nullable = false)
+    String text;
```
Добавь, пожалуйста, private модификатор доступа.

Chegulov replied:
спасибо, добавил.

avlin87 reviewed (changes requested):
Привет, Алексей,
Работа выглядит хорошо, автоматические проверки проходят успешно.
Оставил 3 комментария, смотри, пожалуйста, ниже.

Chegulov committed:
add-bookings
добавлена метод createDtoOutputList(List<Booking> bookings) в BookingMapper.
приведен state к UpperCase.
добавлен private модификатор к полю Comment.text

avlin87 reviewed (approved):
Работу принимаю.
